src/bootstrap/src/core/builder.rs

```diff
@@ -2094,11 +2094,9 @@ impl<'a> Builder<'a> {
         }
 
         // set rustc args passed from command line
-        let rustc_args =
-            self.config.cmd.rustc_args().iter().map(|s| s.to_string()).collect::<Vec<_>>();
-        if !rustc_args.is_empty() {
-            cargo.env("RUSTFLAGS", &rustc_args.join(" "));
-        }
+        self.config.cmd.rustc_args().iter().for_each(|v| {
+            rustflags.arg(v);
+        });
```

Comment:
The --help text says this should only be passed when running tests; I don't see that reflected in the code. So which one is right and which one is wrong?

Reply:
It is reflected in the code: https://github.com/rust-lang/rust/blob/c0ddaef075748674140263840a185082f44458e9/src/bootstrap/src/core/config/flags.rs#L462-L469

I will make the comment more clear.

Reply:
> I will make the comment more clear.

Updated: https://github.com/rust-lang/rust/compare/69a145ad0a472dcbf40665cec5096053240e8bb0..199589d81466a4b436bd41cc0dfd2f35c908a951